### PR TITLE
chore: upgrade django 4.2 LTS (@W-14178534)

### PR DESCRIPTION
**Pull request by rohansroy**
- [x] Update CHANGELOG.md
- [x] Update README.md (as needed)
- [x] New dependencies were added to `pyproject.toml`
- [x] `pip install` succeeds with a clean virtualenv
- [x] There are new or modified tests that cover changes
- [x] Test coverage is maintained or expanded

### CODE REVIEW

**Comment by bgrant** [comment]
Would you mind adding to the CHANGELOG.md?

**Comment by bgrant** [comment]
Testing with our service often uncovers problems with DDA changes that the unit tests don't.  I'll leave it to you if you want to go ahead and merge this change or wait until you do that testing.

**Review by bgrant — CHANGES_REQUESTED** [review]
Can you add a line to the CHANGELOG.md?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [PR 144](https://github.com/salesforce/django-declarative-apis/pull/141) Allow Dependabot to update dev dependencies
 - [PR 143](https://github.com/salesforce/django-declarative-apis/pull/143) Testing: Replace `flake8` and `black` with `ruff`, add testing for Python 3.12, drop testing for Python 3.7
 - [PR 146](https://github.com/salesforce/django-declarative-apis/pull/146) Add ExpandableGeneric filter type
+- [PR 148](https://github.com/salesforce/django-declarative-apis/pull/148) chore: upgrade django 4.2 LTS (@W-14178534)
```
Comment (bgrant) [inline CHANGELOG.md:11]: Thank you for adding this.

It looks like we forgot to update this CHANGELOG the last time we released DDA.  This release contains the "ExpandableGeneric" filter type (and I assume the PRs before it):

[Release Django Declarative APIs v0.31.6 · salesforce/django-declarative-apis](https://github.com/salesforce/django-declarative-apis/releases/tag/v0.31.6)

This isn't your mistake, but would you mind fixing the CHANGELOG?

Comment (bgrant) [reply CHANGELOG.md:11]: You could potentially fix it in the PR when you make a release, so I'll go ahead and approve this PR.

**Review by bgrant — APPROVED** [review]
I found an issue with the CHANGELOG.  Would you mind fixing it when you update the CHANGELOG as part of making a new release?

**Comment by rohansroy** [comment]
> I found an issue with the CHANGELOG. Would you mind fixing it when you update the CHANGELOG as part of making a new release?

will do, to save time i'll merge this and fix the changelog with the release PR.

**Pull request merged**